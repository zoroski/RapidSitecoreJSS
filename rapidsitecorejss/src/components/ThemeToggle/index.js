import React from 'react';
import { withSitecoreContext } from '@sitecore-jss/sitecore-jss-react'

const themes = ['primary', 'minor'];

const ThemeToggle = (props) => {
  const { sitecoreContext } = props;

  // actually it's bad, temporary & doesn't work as u expect. Need to extend context setter to update the app after updating the context
  if (!sitecoreContext.theme) {
    sitecoreContext.theme = themes[0];
  }

  const change = (newTheme) => {
    console.log(sitecoreContext.theme, newTheme)
    sitecoreContext.theme = newTheme;
  };

  return (
    <div>
      <p>Current theme: {sitecoreContext.theme}</p>
      {
        themes.map(theme => 
          <label key={theme} style={{ marginRight: '10px '}}>
            {theme}
            <input type="radio" name="theme" value={theme} checked={theme === sitecoreContext.theme} onChange={() => change(theme)} style={{ marginLeft: '5px' }} />
          </label>
        )
      }
    </div>
  )
};

export default withSitecoreContext()(ThemeToggle);
