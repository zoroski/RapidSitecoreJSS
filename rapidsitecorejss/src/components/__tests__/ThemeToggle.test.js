import React from 'react';
import ThemeToggle from '../ThemeToggle/index';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
      const tree = renderer
        .create( < ThemeToggle fields = {
            {
              heading: {
                value: 'Heading',
              },
            }
          }
          />)
          .toJSON(); expect(tree).toMatchSnapshot();
        });