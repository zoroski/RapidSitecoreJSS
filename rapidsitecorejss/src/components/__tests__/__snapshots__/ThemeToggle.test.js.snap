// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <p>
    Toggle theme
  </p>
  <input
    name="toggle"
    onChange={[Function]}
    type="radio"
  />
  <input
    name="toggle"
    onChange={[Function]}
    type="radio"
  />
  Heading
</div>
`;
